aiohttp/client_reqrep.py

```diff
@@ -70,7 +70,9 @@
 try:
     import cchardet as chardet
 except ImportError:  # pragma: no cover
-    import chardet  # type: ignore[no-redef]
+    import charset_normalizer as chardet  # type: ignore[no-redef]
+
+    warnings.filterwarnings('ignore', 'Trying to detect', module='charset_normalizer')
```

Comment:
I'm curious about this. Why are we ignoring a warning? If there's nothing the user can do in regards to the warning, then it probably shouldn't be a warning in the upstream library (but, logging.warning instead).

Reply:
Good question. 

> Why are we ignoring a warning? If there's nothing the user can do in regards to the warning, then it probably shouldn't be a warning.

You are right about the main logic behinds warnings.

But its a little bit trickier in that case.

The warning targeted here is when a tiny content (less than 32o) is given for detection. Sometime its reliable, sometime not. Those cases are often tiny JSON / ASCII / u8 content in my experiences.

In some ways, the end user can do something about it but often wont. eg. server not adding the explicit `charset` when it obviously should.

It is entirely up to aiohttp maintainers to choose whether or not a warning should be raised in that context.
It could potentially spam the stdout and raises complaints.

I am happy to debate around the necessity of the raised warning inside this library or even in `charset_normalizer`.

Reply:
> In some ways, the end user can do something about it but often wont. eg. server not adding the explicit charset when it obviously should.

But, if it's mainly going to be used for decoding data that is sent to the user, in most cases, the user will have no control to fix the source.

If we wanted to warn users about cases where the charset is not set, then we'd do that in our code every time charset_normalizer is invoked. Making it dependant on the size of the input seems less useful.

If you have control over the source, in order to fix the warning, why don't you know the encoding already?

Reply:
Frankly I am tempted to remove the warning in favour of a straight forward logging.warning instead.
In a that environment, I would totally agree. I am considering to remove the warning in favour of a simple logging output.
Thanks for your inputs.

Meanwhile, would this be a blocker?

Reply:
No, the blocker is somebody else more familiar with this giving an approval.

Reply:
OK. Will wait upon that.

Reply:
I have removed the "silent" warning set for `charset_normalizer`. The release 2.0.5 no longer raise any warning.
From what I have seen in the project (commits&code owners), the best candidate to address the approval is asvetlov.

Reply:
I agree, but might be awhile before he has a chance to look over these PRs.